### PR TITLE
Adds Superhuman Physiology, a martial art used by clings.

### PR DESCRIPTION
**Pull request by hal9000PR**
## What Does This PR Do
Adds a new martial art, used by clings it can be purchased for 3 DNA points
this martial art has a couple properties,

If you are examined by someone within one tile of you (similarly to shadowling thralls) they will get an examine message denoting that you have this martial art. `Their muscles seem to buldge and shift in an inhuman manner.`

**basic attacks**

harm intent: the cling uses claws to lash at their target dealing 12 damage per swing

disarm:  works slightly more often, and knocks the target back one tile when successful doing a little bit of damage

grab: instantly aggressive grab

**combos** 

Hammer fist  
Steps: Disarm, Harm, Harm

Deals 20 damage, stuns the target for 2-4 seconds, and deals 30 stamina damage deals more damage if the target is already stunned. Doesn't stun targets who are already stunned.

*We morph our fist into a bone hammer and slam our target to the ground, deals double damage to already stunned targets.*

Piston Kick
Steps: Disarm, Grab

deals little damage but launches the target 10 tiles away and confuses them, good for disengaging. also damages one of your legs a small amount

*We use explosive chemical reactions to kick someone with such a force that it sends them flying*

Foot Skewer
Steps: Harm, Grab

deals damage to your targets foot, and slows them for 10 seconds, good opening move, makes clicking on people easier.

*We reform our foot into a knife and stab it through our opponent's foot, crippling their mobility.*

Brutal Barrage
Steps: Harm, Disarm, Grab, Harm

Punches your opponent in the face really hard, deals more damage the more damage the target has already taken. good finishing move.

*We unleash a flurry of swift blows that completely decimate a weakened opponent.*
## Why It's Good For The Game

Currently, clings are a little lacklustre. you are better off not using changeling abilities because of their KOS status. This adds a little more variety to clings, instead of it just being "hurr durr armblade" there is more depth to it than just clicking until horizontal.

Personally I think this suits the cling aesthetic as well. Using their advanced biology and memories/skills of those who they have once absorbed to form a complex system of attacks, growing claws and knives, using their heightened strength to knock people down.

The damage and stun numbers are entirely changeable if you guys think they are too high or too low.

## Changelog
:cl:
add: adds a new martial art purchasable by clings for 3 DNA points, Superhuman physiology. Basic punches are claws, dealing high damage. Disarms are more likely to work and knockback and damage the target. Being examined within one tile gives the examiner a message denoting that you have the ability, similarly to shadowling thralls. you get 4 combos aswell, hammer fist - a high damage stunning combo. Piston kick - a knockback combo. Foot skewer - a light damage combo which slows the target. Brutal barrage - a combo that does more damage the more damaged the target is.

/:cl:

### CODE REVIEW

File: code/game/gamemodes/changeling/powers/superhuman_physiology.dm [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/datum/action/changeling/superhuman
+	name = "Superhuman Physiology"
+	helptext = "We may toggle our superhuman biology on and off to avoid detection."
```
Comment (S34NW) [inline code/game/gamemodes/changeling/powers/superhuman_physiology.dm:3]:
```suggestion
	helptext = "We can toggle our superhuman biology on and off to avoid detection."
```

File: code/game/gamemodes/changeling/powers/superhuman_physiology.dm [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	desc = "We combine the fighting knowledge of the hivemind and our enhanced biological strength to form a potent martial art that makes use of our unique abilty to create weapons. Costs 30 chemicals to toggle"
```
Comment (S34NW) [inline code/game/gamemodes/changeling/powers/superhuman_physiology.dm:4]:
```suggestion
	desc = "We combine the fighting knowledge of the hivemind and our enhanced biological strength to form a potent martial art that makes use of our unique abilty to create weapons. Costs 30 chemicals to toggle."
```

File: code/modules/martial_arts/combos/superhuman/piston_kick.dm [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/datum/martial_combo/superhuman/piston_kick
+	name = "Piston Kick"
+	explaination_text = "We use explosive chemical reactions to kick someone with such a force that it sends them flying"
```
Comment (S34NW) [inline code/modules/martial_arts/combos/superhuman/piston_kick.dm:3]:
```suggestion
	explaination_text = "We use explosive chemical reactions to kick someone with such a force that it sends them flying."
```

File: code/modules/martial_arts/combos/superhuman/piston_kick.dm [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+	steps = list(MARTIAL_COMBO_STEP_DISARM, MARTIAL_COMBO_STEP_GRAB)
+
+/datum/martial_combo/superhuman/piston_kick/perform_combo(mob/living/carbon/human/user, mob/living/target, datum/martial_art/MA)
+	user.do_attack_animation(target, ATTACK_EFFECT_KICK)
+	user.apply_damage(5, BURN, pick(BODY_ZONE_R_LEG, BODY_ZONE_L_LEG)) //your leg is literally exploding like a piston
+	var/atom/throw_target = get_edge_target_turf(target, user.dir)
+	target.throw_at(throw_target, 10, 14, user)
+	target.visible_message("<span class='warning'>[user]'s lower leg explosively extends into [target] sending them flying!</span>", \
```
Comment (S34NW) [inline code/modules/martial_arts/combos/superhuman/piston_kick.dm:11]:
```suggestion
	target.visible_message("<span class='warning'>[user]'s lower leg is explosively propelled into [target], sending them flying!</span>", \
```

**Comment by Samman166** [comment]
This sounds cool as fuck and I want it. On one hand I think it might make armblade a bit obsolete but it still does double the damage as this and is useful just to pry doors/remove heads

File: code/modules/martial_arts/combos/superhuman/piston_kick.dm [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+					"<span class='userdanger'>[user] lower leg explosively extends into you! sending you flying and disorintating you!</span>")
```
Comment (S34NW) [inline code/modules/martial_arts/combos/superhuman/piston_kick.dm:12]:
```suggestion
					"<span class='userdanger'>[user]'s lower leg is explosively propelled into you! Sending you flying and disorienting you!</span>")
```

File: code/modules/martial_arts/superhuman.dm [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+		playsound(D, 'sound/weapons/punchmiss.ogg', 25, 1, -1)
+
+	add_attack_logs(A, D, "Melee attacked with martial-art [src] : Disarmed", ATKLOG_ALL)
+	return TRUE
+
+/datum/martial_art/superhuman/explaination_header(user)
+	to_chat(usr, "<b><i>We scour the memories of all those we have absorbed, remembering their fighting styles... </i></b>")
+
+/datum/martial_art/superhuman/explaination_footer(user)
+	to_chat(user, "<b><i>In addition, by having your throw mode on when being attacked, we enter an active defense mode where we block and sometimes even counter attacks done to you.</i></b>")
```
Comment (S34NW) [inline code/modules/martial_arts/superhuman.dm:51]:
```suggestion
	to_chat(user, "<b><i>In addition, by having our throw mode on when being attacked, we enter an active defense mode where we block and sometimes even counter melee attacks done to us.</i></b>")
```

**Review by S34NW — COMMENTED** [review]
Really good work, just some small grammar things.

File: code/game/gamemodes/changeling/powers/superhuman_physiology.dm (hunk 1 above)
Comment (hal9000PR) [reply code/game/gamemodes/changeling/powers/superhuman_physiology.dm:3]: wait.... thats not even something you can do. description is just wrong here

File: code/modules/martial_arts/combos/superhuman/brutal_barrage.dm [hunk 7]
```diff
@@ -0,0 +1,17 @@
+/datum/martial_combo/superhuman/brutal_barrage
+	name = "Brutal Barrage"
+	explaination_text = "We unleash a flurry of swift blows that completely decimate a weakened oppenent."
```
Comment (LordShekelstein) [inline code/modules/martial_arts/combos/superhuman/brutal_barrage.dm:3]:
```suggestion
	explaination_text = "We unleash a flurry of swift blows that completely decimates a weakened oppenent."
```
haha u luv martial orts lmaoooo

File: code/modules/martial_arts/combos/superhuman/foot_skewer.dm [hunk 8]
```diff
@@ -0,0 +1,16 @@
+/datum/martial_combo/superhuman/foot_skewer
+	name = "Foot Skewer"
+	explaination_text = "We reform our foot into a knife and stab it through our opponent's foot, crippling their mobility."
```
Comment (LordShekelstein) [inline code/modules/martial_arts/combos/superhuman/foot_skewer.dm:3]:
```suggestion
	explaination_text = "We reform our foot into a serrated blade and stab it deep into our opponent's foot eviscerating tendons and ligaments, crippling their mobility.
."
```
This is mostly fluff text to make it more epic. Your grammar and spelling was correct.

Comment (LordShekelstein) [reply code/modules/martial_arts/combos/superhuman/foot_skewer.dm:3]: Also my quote is in the wrong spot, heck.

File: code/modules/martial_arts/superhuman.dm [hunk 9]
```diff
@@ -0,0 +1,51 @@
+/datum/martial_art/superhuman
+	name = "Superhuman Physiology"
+	block_chance = 100
+	has_explaination_verb = TRUE
+	combos = list(/datum/martial_combo/superhuman/hammer_fist, /datum/martial_combo/superhuman/piston_kick, /datum/martial_combo/superhuman/foot_skewer, /datum/martial_combo/superhuman/brutal_barrage)
+
+/datum/martial_art/superhuman/grab_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	MARTIAL_ARTS_ACT_CHECK
+	var/obj/item/grab/G = D.grabbedby(A, 1)
```
Comment (SabreML) [inline code/modules/martial_arts/superhuman.dm:9]:
```suggestion
	var/obj/item/grab/G = D.grabbedby(A, TRUE)
```

File: code/modules/martial_arts/combos/superhuman/foot_skewer.dm (hunk 8 above)
Comment (LordShekelstein) [reply code/modules/martial_arts/combos/superhuman/foot_skewer.dm:3]: Mhm, apologies, I had a major migraine whilst writing my changes.

File: code/modules/martial_arts/superhuman.dm [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+/datum/martial_art/superhuman/disarm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	MARTIAL_ARTS_ACT_CHECK
+	A.do_attack_animation(D, ATTACK_EFFECT_DISARM)
+	if(prob(60))
+		if(!D.stat && !D.IsWeakened())
+			D.visible_message("<span class='warning'>[A] slams into [D], knocking them to the ground!</span>", \
+								"<span class='userdanger'>[A] slams into you, knocking you to the ground!</span>")
+			playsound(get_turf(D), 'sound/weapons/punchmiss.ogg', 50, 1, -1)
+			D.apply_damage(5, BRUTE)
+			D.Weaken(2)
```
Comment (SabreML) [inline code/modules/martial_arts/superhuman.dm:36]: 60% chance to stun for 4 seconds every time you disarm seems pretty high to me.

File: code/modules/martial_arts/superhuman.dm [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+			if(A != D)
+				var/atom/throw_target = get_edge_target_turf(D, A.dir)
+				D.throw_at(throw_target, 1, 14, A)
+	else
+		D.visible_message("<span class='danger'>[A] attempted to disarm [D]!</span>", "<span class='userdanger'>[A] attempted to disarm [D]!</span>")
```
Comment (SabreML) [inline code/modules/martial_arts/superhuman.dm:41]:
```suggestion
		D.visible_message("<span class='danger'>[A] attempted to disarm [D]!</span>", "<span class='userdanger'>[A] attempted to disarm you!</span>")
```
?

File: code/modules/martial_arts/combos/superhuman/brutal_barrage.dm [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+/datum/martial_combo/superhuman/brutal_barrage/perform_combo(mob/living/carbon/human/user, mob/living/target, datum/martial_art/MA)
+	user.do_attack_animation(target, ATTACK_EFFECT_SMASH)
+	playsound(get_turf(target), 'sound/weapons/punch1.ogg', 75, 1, -1)
+	if(target.health <= 20)
+		target.visible_message("<span class='warning'>[user]'s arm swings, enlarging before impact, striking [target]'s face with great force, knocking them out!</span>", \
+						  "<span class='userdanger'>[user] slams you in the face, knocking you out!</span>")
+		target.SetSleeping(10) //KO
+	else
+		target.visible_message("<span class='warning'>[user]'s arm swings, enlarging before impact, striking [target]'s face with great force!</span>", \
+						  "<span class='userdanger'>[user] slams you in the face!</span>")
```
Comment (SabreML) [inline code/modules/martial_arts/combos/superhuman/brutal_barrage.dm:15]:
```suggestion
		target.visible_message("<span class='warning'>[user]'s arm swings, enlarging before impact, striking [target]'s face with great force, knocking them out!</span>",
						  "<span class='userdanger'>[user] slams you in the face, knocking you out!</span>")
		target.SetSleeping(10) //KO
	else
		target.visible_message("<span class='warning'>[user]'s arm swings, enlarging before impact, striking [target]'s face with great force!</span>",
						  "<span class='userdanger'>[user] slams you in the face!</span>")
```
You don't actually need the backslashes here after the `visible_message()`. (Even though they're _everywhere_ in the codebase.)
Those would only apply if you wanted to continue the string onto the next line without cutting it. In this case you've finished with the first argument so the comma makes it perfectly fine for the second argument to be on a new line.
Same thing in the other files too.

**Review by SabreML — COMMENTED** [review]
👀

File: code/modules/martial_arts/combos/superhuman/hammer_fist.dm [hunk 13]
```diff
@@ -0,0 +1,17 @@
+/datum/martial_combo/superhuman/hammer_fist
+	name = "Hammer Fist"
+	explaination_text = "We morph our fist into a bone hammer and slam our target to the ground, deals double damage to already stunned targets."
```
[the remark this answers is not in the document]
Comment (Wojz) [reply code/modules/martial_arts/combos/superhuman/hammer_fist.dm:3]: I think it might be even better as "The damage is doubled against already stunned targets."

**Comment by Wojz** [comment]
100% block on an antag with virtually unlimited EMP might be a tad strong, the baton is frequently the way to deal with the buggers

**Comment by Rurik123** [comment]
> grab: instantly aggressive grab

This'll be fun.

Though it should be noted, this will make mindswaps even easier. A literal naked guy can instant grab HoS walking down the hallway, swapping bodies before help can arrive. I don't mind too much since usually victim gets a call of help and bystanders are almost always around, but figured it should be said.

**Comment by procdrone** [comment]
im little concerned for ability to reliably stun & deal massive damage for free but looks cool otherwise

**Comment by Wojz** [comment]
> > grab: instantly aggressive grab
> A literal naked guy can instant grab HoS walking down the hallway, swapping bodies before help can arrive. 

Did you mean: _**Tentacle**_?

File: code/game/gamemodes/changeling/powers/superhuman_physiology.dm (hunk 1 above)
Comment (hal9000PR) [reply code/game/gamemodes/changeling/powers/superhuman_physiology.dm]: Fixed

**Comment by Dumbdumn5** [comment]
I love this, but what're the stun times in plain english because I am n o t a code wizard and don't quite understand setting somebody's 'sleep' to ten. Is it ten ingame seconds of unconsciousness? Because that's asinine in terms of power and combat changes are trying to r e d u c e stuns. 

When looking at numbers in combat, one second is a minor hindrance, three seconds is a serious issue, and five seconds is an eternity. Considering this martial art seems to be about zoning and combo-mealing people into oblivion, keeping the times quick rather than very long should help. That ten tile fling and confuse should do enough to stun people on its own, just sorta curious on the hammer stun time in seconds is all.

**Review by Kyep — COMMENTED** [review]
Frankly, this seems a bit OP. Rather like arm blade, except mostly better, and people can't tell you have it (and thus, are a cling) from looking at your sprite.

**Comment by hal9000PR** [comment]
Right, imma reduce some of the stun times but maybe I'll substitute some of them with stamina damage? So initially the stuns are short and you can escape/fight back. But after a while of using combos on someone they get stunned for much longer? 

 @Dumbdumn5 `weaken(3)` means it will stun someone for 3 cycles, meaning 4-6 seconds, so basically a telebaton. The sleep means 18-20 seconds which admittedly could be a little long, however it is a lengthy combo which involves a grab, making it harder to pull off because you cannot do it on the move. Once you grab them you are slowed, so having a long difficult combo to pull off which does big damage and if you are hurt already knocks you out could be okay. However if you feel any of this is too much i can: decrease the stun times, decrease the damage values, increase the DNA cost.

File: code/game/gamemodes/changeling/powers/superhuman_physiology.dm [hunk 14]
```diff
@@ -0,0 +1,15 @@
+/datum/action/changeling/superhuman
+	name = "Superhuman Physiology"
+	desc = "We combine the fighting knowledge of the hivemind and our enhanced biological strength to form a potent martial art that makes use of our unique abilty to create weapons."
+	dna_cost = 3
+	req_human = 0
```
Comment (farie82) [inline code/game/gamemodes/changeling/powers/superhuman_physiology.dm:5]:
```suggestion
	req_human = FALSE
```

File: code/modules/martial_arts/combos/superhuman/brutal_barrage.dm [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+	user.do_attack_animation(target, ATTACK_EFFECT_SMASH)
+	playsound(get_turf(target), 'sound/weapons/punch1.ogg', 75, TRUE, -1)
+	if(target.health <= 0)
+		target.visible_message("<span class='warning'>[user]'s arm swings, enlarging before impact, striking [target]'s face with great force, knocking them out!</span>",
+						  "<span class='userdanger'>[user] slams you in the face, knocking you out!</span>")
+		target.SetSleeping(5) //KO
+	else
+		target.visible_message("<span class='warning'>[user]'s arm swings, enlarging before impact, striking [target]'s face with great force!</span>",
+						  "<span class='userdanger'>[user] slams you in the face!</span>")
+	target.apply_damage(target.getBruteLoss() * 0.75, BRUTE, BODY_ZONE_HEAD)
```
Comment (farie82) [inline code/modules/martial_arts/combos/superhuman/brutal_barrage.dm:16]: This can do an absurd amount of damage if left uncapped. Just a note

File: code/modules/martial_arts/combos/superhuman/brutal_barrage.dm [hunk 16]
```diff
@@ -0,0 +1,17 @@
+/datum/martial_combo/superhuman/brutal_barrage
+	name = "Brutal Barrage"
+	explaination_text = "We unleash a flurry of swift blows that completely decimates an oppenent in a critical state."
+	steps = list(MARTIAL_COMBO_STEP_HARM, MARTIAL_COMBO_STEP_DISARM, MARTIAL_COMBO_STEP_GRAB, MARTIAL_COMBO_STEP_HARM)
+
+/datum/martial_combo/superhuman/brutal_barrage/perform_combo(mob/living/carbon/human/user, mob/living/target, datum/martial_art/MA)
```
Comment (farie82) [inline code/modules/martial_arts/combos/superhuman/brutal_barrage.dm:6]: This one does not have an `add_attack_logs`

File: code/modules/martial_arts/combos/superhuman/foot_skewer.dm [hunk 17]
```diff
@@ -0,0 +1,16 @@
+/datum/martial_combo/superhuman/foot_skewer
+	name = "Foot Skewer"
+	explaination_text = "We reform our foot into a serrated blade and stab it deep into our opponent's foot, eviscerating tendons and ligaments and crippling their mobility."
+	steps = list(MARTIAL_COMBO_STEP_HARM, MARTIAL_COMBO_STEP_GRAB)
+
+/datum/martial_combo/superhuman/foot_skewer/perform_combo(mob/living/carbon/human/user, mob/living/target, datum/martial_art/MA)
```
Comment (farie82) [inline code/modules/martial_arts/combos/superhuman/foot_skewer.dm:6]: This one does not have an `add_attack_logs`

File: code/modules/martial_arts/superhuman.dm [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+				D.throw_at(throw_target, 1, 14, A)
+	else
+		D.visible_message("<span class='danger'>[A] attempted to disarm [D]!</span>", "<span class='userdanger'>[A] attempted to disarm you!</span>")
+		playsound(D, 'sound/weapons/punchmiss.ogg', 25, TRUE, -1)
+
+	add_attack_logs(A, D, "Melee attacked with martial-art [src] : Disarmed", ATKLOG_ALL)
+	return TRUE
+
+/datum/martial_art/superhuman/explaination_header(user)
+	to_chat(usr, "<b><i>We scour the memories of all those we have absorbed, remembering their fighting styles... </i></b>")
```
Comment (farie82) [inline code/modules/martial_arts/superhuman.dm:48]:
```suggestion
	to_chat(user, "<b><i>We scour the memories of all those we have absorbed, remembering their fighting styles... </i></b>")
```

**Review by farie82 — CHANGES_REQUESTED** [review]
Got some minor code things.
Glad to see somebody making new martial arts. It does look cool I gotta admit.

I do have a feeling that this might be a bit strong though. Especially the fact it is hidden till used and always activatable in combination with the burst like combos.
Maybe add more distinguishable effects to the attacks for players to pick up on? Or add effects when the stance is on.

File: code/modules/martial_arts/combos/superhuman/brutal_barrage.dm [hunk 19]
```diff
@@ -0,0 +1,17 @@
+/datum/martial_combo/superhuman/brutal_barrage
+	name = "Brutal Barrage"
+	explaination_text = "We unleash a flurry of swift blows that completely decimates an oppenent in a critical state."
+	steps = list(MARTIAL_COMBO_STEP_HARM, MARTIAL_COMBO_STEP_DISARM, MARTIAL_COMBO_STEP_GRAB, MARTIAL_COMBO_STEP_HARM)
```
Comment (farie82) [inline code/modules/martial_arts/combos/superhuman/brutal_barrage.dm:4]: One more small thing. Having grab and then something afterwards will give a wrong help text. Since using grab and then harm on somebody will do nothing. Since you now have a grab as your hand. Bit annoying thing and people will get confused why it is not working while they follow the instructions to the point

File: code/modules/martial_arts/superhuman.dm (hunk 10 above)
Comment (hal9000PR) [reply code/modules/martial_arts/superhuman.dm]: This is fairly similar to standard disarms. Also it is a 2-4 second stun rather than 4 seconds. I'm unsure on how to make the disarms better without completely reworking disarms in general.

**Comment by Dashsora** [comment]
Overall, I absolutely love it, and it sounds terribly fun, but I think the instant aggressive grab raises some concerns. 

It renders tentacle kind of pointless, as, yes, it's ranged, but it's also extremely unreliable. I would very often rather attempt to close in the gap, or I'm in melee range already. It doesn't have the rest of its utility, but it has *so much more*, and it doesn't restrict you from using long range weapons either.

I've already told Charlie my suggestion for this, but in the interest to further the conversation here as well, I'd suggest making getting an aggressive grab out of a combo - the death hug.

Help intent - Grab. "We disguise our approach with a hug !"

Not only is it more healthy mechanically, I think it's also terribly thematic to clings as being antags who get you to fear everyone around you. Once there are clings announced on station, watch everyone start to be scared of the hug of death !

**Comment by hal9000PR** [comment]
Update: removed the block chance from the martial art
Added a line of text when you examine someone, when one tile away from them, who knows the martial art, denoting that they have it `Their muscles seem to buldge and shift in an inhuman manner.` 

Hopefully this clears up some of @Kyep 's concerns about it not being visible enough. Now at any point a random person can examine you if you pick this and you get called out as a cling, similar to cling eyes and flashes. So it might not be as blatant as arm blade when it is actually being used (even though it is rather obvious, normal people don't tend to claw and smash people) but it is constantly semi-visible. So picking this option is a risk, you have to always position yourself more than one tile away from people to not be examined and given away. AND it costs more than armblade in DNA points.

File: code/game/gamemodes/changeling/powers/superhuman_physiology.dm [hunk 20]
```diff
@@ -0,0 +1,15 @@
+/datum/action/changeling/superhuman
+	name = "Superhuman Physiology"
+	desc = "We combine the fighting knowledge of the hivemind and our enhanced biological strength to form a potent martial art that makes use of our unique abilty to create weapons."
```
Comment (Sirryan2002) [inline code/game/gamemodes/changeling/powers/superhuman_physiology.dm:3]:
```suggestion
	desc = "We utilize the pooled knowledge of the hivemind and our enhanced physiology to further weaponize our unique abilities into a dynamic fighting style."
```

File: code/modules/martial_arts/combos/superhuman/foot_skewer.dm [hunk 21]
```diff
@@ -0,0 +1,17 @@
+/datum/martial_combo/superhuman/foot_skewer
+	name = "Foot Skewer"
+	explaination_text = "We reform our foot into a serrated blade and stab it deep into our opponent's foot, eviscerating tendons and ligaments and crippling their mobility."
```
Comment (Sirryan2002) [inline code/modules/martial_arts/combos/superhuman/foot_skewer.dm:3]:
```suggestion
	explaination_text = "We reform our leg into a serrated blade and jab it deep into our opponent's foot, piercing tendons and ligaments, crippling their mobility."
```
better phrasing and vocab

File: code/modules/martial_arts/combos/superhuman/foot_skewer.dm [hunk 22]
```diff
@@ -0,0 +1,17 @@
...
+	steps = list(MARTIAL_COMBO_STEP_HARM, MARTIAL_COMBO_STEP_GRAB)
+
+/datum/martial_combo/superhuman/foot_skewer/perform_combo(mob/living/carbon/human/user, mob/living/target, datum/martial_art/MA)
+	if(!target.stat && !target.IsWeakened())
+		user.do_attack_animation(target, ATTACK_EFFECT_CLAW)
+		playsound(get_turf(target), 'sound/weapons/slash.ogg', 50, TRUE, -1)
+		target.visible_message("<span class='warning'>[user] reforms their foot into a blade and stabs it into [target]'s foot!</span>",
```
Comment (Sirryan2002) [inline code/modules/martial_arts/combos/superhuman/foot_skewer.dm:10]:
```suggestion
		target.visible_message("<span class='warning'>[user] reforms their foot into a blade and jabs it deep into [target]'s foot!</span>",
```

File: code/modules/martial_arts/combos/superhuman/foot_skewer.dm [hunk 23]
```diff
@@ -0,0 +1,17 @@
...
+						  "<span class='userdanger'>[user] stabs you in the foot, crippling your ability to walk!</span>")
```
Comment (Sirryan2002) [inline code/modules/martial_arts/combos/superhuman/foot_skewer.dm:11]:
```suggestion
						  "<span class='userdanger'>[user] reforms [user.p_their()] foot into a blade and sinks it deep into your foot, crippling your ability to walk!</span>")
```

File: code/modules/martial_arts/combos/superhuman/hammer_fist.dm [hunk 24]
```diff
@@ -0,0 +1,18 @@
+/datum/martial_combo/superhuman/hammer_fist
+	name = "Hammer Fist"
+	explaination_text = "We morph our fist into a bone hammer and slam our target to the ground. The damage is doubled against already stunned targets."
```
Comment (Sirryan2002) [inline code/modules/martial_arts/combos/superhuman/hammer_fist.dm:3]:
```suggestion
	explaination_text = "We morph our fist into a bone hammer and slam our target into the ground. Already stunned targets receive twice as much damage."
```

File: code/modules/martial_arts/combos/superhuman/piston_kick.dm [hunk 25]
```diff
@@ -0,0 +1,18 @@
...
+	explaination_text = "We use explosive chemical reactions to kick someone with such a force that it sends them flying."
+	steps = list(MARTIAL_COMBO_STEP_DISARM, MARTIAL_COMBO_STEP_GRAB)
+
+/datum/martial_combo/superhuman/piston_kick/perform_combo(mob/living/carbon/human/user, mob/living/target, datum/martial_art/MA)
+	if(user != target)
+		user.do_attack_animation(target, ATTACK_EFFECT_KICK)
+		user.apply_damage(5, BURN, pick(BODY_ZONE_R_LEG, BODY_ZONE_L_LEG)) //your leg is literally exploding like a piston
+		var/atom/throw_target = get_edge_target_turf(target, user.dir)
+		target.throw_at(throw_target, 10, 14, user)
+		target.visible_message("<span class='warning'>[user]'s lower leg is explosively propelled into [target], sending them flying!</span>",
```
Comment (Sirryan2002) [inline code/modules/martial_arts/combos/superhuman/piston_kick.dm:12]:
```suggestion
		target.visible_message("<span class='warning'>[user]'s lower leg is explosively propelled into [target], sending [target.p_them()] flying!</span>",
```

File: code/modules/martial_arts/combos/superhuman/piston_kick.dm [hunk 26]
```diff
@@ -0,0 +1,18 @@
...
+						"<span class='userdanger'>[user]'s lower leg is explosively propelled into you! Sending you flying and disorienting you!</span>")
```
Comment (Sirryan2002) [inline code/modules/martial_arts/combos/superhuman/piston_kick.dm:13]:
```suggestion
						"<span class='userdanger'>[user]'s lower leg is explosively propelled into you, disorienting you and sending you flying!</span>")
```

File: code/modules/martial_arts/superhuman.dm [hunk 27]
```diff
@@ -0,0 +1,52 @@
...
+	playsound(get_turf(D), 'sound/weapons/slice.ogg', 25, TRUE, -1)
+	add_attack_logs(A, D, "Melee attacked with martial-art [src] : Punched", ATKLOG_ALL)
+	return TRUE
+
+/datum/martial_art/superhuman/disarm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	MARTIAL_ARTS_ACT_CHECK
+	A.do_attack_animation(D, ATTACK_EFFECT_DISARM)
+	if(prob(60))
+		if(!D.stat && !D.IsWeakened())
+			D.visible_message("<span class='warning'>[A] slams into [D], knocking them to the ground!</span>",
```
Comment (Sirryan2002) [inline code/modules/martial_arts/superhuman.dm:31]:
```suggestion
			D.visible_message("<span class='warning'>[A] slams into [D], knocking [D.p_them()] to the ground!</span>",
```

File: code/modules/martial_arts/superhuman.dm [hunk 28]
```diff
@@ -0,0 +1,52 @@
...
+
+	add_attack_logs(A, D, "Melee attacked with martial-art [src] : Disarmed", ATKLOG_ALL)
+	return TRUE
+
+/datum/martial_art/superhuman/explaination_header(user)
+	to_chat(user, "<b><i>We scour the memories of all those we have absorbed, remembering their fighting styles... </i></b>")
+
+/datum/martial_art/superhuman/teach(mob/living/carbon/human/H, make_temporary)
+	. = ..()
+	to_chat(H, "<span class='sciradio'> We shift our biology, we are now able to rapidly form weapons out of different limbs after preforming different attacks. \
```
Comment (Sirryan2002) [inline code/modules/martial_arts/superhuman.dm:51]:
```suggestion
	to_chat(H, "<span class='sciradio'> We alter our biology, we are now able to rapidly form weapons out of different limbs when preforming special attacks. \
```
better phrasing

File: code/modules/martial_arts/superhuman.dm [hunk 29]
```diff
@@ -0,0 +1,52 @@
...
+				Our basic punches are augmented to be slashing claws that do large damage at the cost of knockdown chance. Our disarms become aggressive slams that ram our target to the floor. <span/>")
```
Comment (Sirryan2002) [inline code/modules/martial_arts/superhuman.dm:52]:
```suggestion
				When punching, our fingers become slashing claws that shred flesh. However, it is at the cost of lower knockdown chance. Our disarms aggressively slam opponents into the floor. <span/>")
```

**Review by Sirryan2002 — CHANGES_REQUESTED** [review]

File: code/modules/martial_arts/superhuman.dm [hunk 30]
```diff
@@ -0,0 +1,52 @@
...
+
+	add_attack_logs(A, D, "Melee attacked with martial-art [src] : Disarmed", ATKLOG_ALL)
+	return TRUE
+
+/datum/martial_art/superhuman/explaination_header(user)
+	to_chat(user, "<b><i>We scour the memories of all those we have absorbed, remembering their fighting styles... </i></b>")
+
+/datum/martial_art/superhuman/teach(mob/living/carbon/human/H, make_temporary)
+	. = ..()
+	to_chat(H, "<span class='sciradio'> We alter our biology, we are now able to rapidly form weapons out of different limbs when preforming special attacks. \
```
Comment (S34NW) [inline code/modules/martial_arts/superhuman.dm:51]:
```suggestion
	to_chat(H, "<span class='sciradio'> We alter our biology, we are now able to rapidly form weapons out of different limbs when performing special attacks. \
```

File: code/modules/martial_arts/combos/superhuman/piston_kick.dm [hunk 31]
```diff
@@ -0,0 +1,18 @@
...
+		user.apply_damage(5, BURN, pick(BODY_ZONE_R_LEG, BODY_ZONE_L_LEG)) //your leg is literally exploding like a piston
+		var/atom/throw_target = get_edge_target_turf(target, user.dir)
+		target.throw_at(throw_target, 10, 14, user)
+		target.visible_message("<span class='warning'>[user]'s lower leg is explosively propelled into [target], sending [target.p_them()] flying!</span>",
+						"<span class='userdanger'>[user]'s lower leg is explosively propelled into you, disorienting you and sending you flying!</span>")
+		add_attack_logs(user, target, "Melee attacked with martial-art [src] : Piston Kick", ATKLOG_ALL)
+		target.Confused(5)
+		target.apply_damage(10, BRUTE)
+		playsound(get_turf(target), 'sound/weapons/punchmiss.ogg', 80, TRUE, -1)
+		return MARTIAL_COMBO_DONE
```
Comment (farie82) [inline code/modules/martial_arts/combos/superhuman/piston_kick.dm:18]:
```suggestion
	if(user == target)
		return MARTIAL_COMBO_FAIL
	user.do_attack_animation(target, ATTACK_EFFECT_KICK)
	user.apply_damage(5, BURN, pick(BODY_ZONE_R_LEG, BODY_ZONE_L_LEG)) //your leg is literally exploding like a 	piston
	var/atom/throw_target = get_edge_target_turf(target, user.dir)
	target.throw_at(throw_target, 10, 14, user)
	target.visible_message("<span class='warning'>[user]'s lower leg is explosively propelled into [target], sending [target.p_them()] flying!</span>",
					"<span class='userdanger'>[user]'s lower leg is explosively propelled into you, disorienting you and sending you flying!</span>")
	add_attack_logs(user, target, "Melee attacked with martial-art [src] : Piston Kick", ATKLOG_ALL)
	target.Confused(5)
	target.apply_damage(10, BRUTE)
	playsound(get_turf(target), 'sound/weapons/punchmiss.ogg', 80, TRUE, -1)
	return MARTIAL_COMBO_DONE
```
Best to apply an early return here and explicitly state that the combo failed

**Review by farie82 — COMMENTED** [review]
Minor thing

**Comment by variableundefined** [comment]
No head or maint approval in 90+ days. Closing.
